sov-modules/sov-modules-api/src/mocks.rs

```diff
@@ -26,21 +25,17 @@ impl MockPublicKey {
     }
 }
 
-impl TryFrom<&str> for MockPublicKey {
-    type Error = Infallible;
-
-    fn try_from(key: &str) -> Result<Self, Self::Error> {
+impl From<&str> for MockPublicKey {
+    fn from(key: &str) -> Self {
         let key = key.as_bytes().to_vec();
-        Ok(Self { pub_key: key })
+        Self { pub_key: key }
     }
 }
 
-impl TryFrom<String> for MockPublicKey {
-    type Error = Infallible;
-
-    fn try_from(key: String) -> Result<Self, Self::Error> {
+impl From<String> for MockPublicKey {
+    fn from(key: String) -> Self {
         let key = key.as_bytes().to_vec();
-        Ok(Self { pub_key: key })
+        Self { pub_key: key }
     }
 }
```

Comment:
non-blocking: It seems like we should be able to deduplicate these by using `AsRef<str>`. Is that right?

Reply:
Good point, done